### ISSUE TITLE
SerializerDataCollector bugs

### ISSUE THREAD

**Issue by b1rdex** [symfony/symfony#46589]
### Symfony version(s) affected

6.1

### Description

**:one: The first thing**:
Create a simple controller with an exception:
```php
#[Route('/test')]
public function test(SerializerInterface $serializer)
{
    $json = $serializer->serialize([], 'json');
    throw new \Exception();
}
```
Open it with a browser, then navigate to Profiler, then to panel Serializer and an error is rendered:
<img width="1041" alt="image" src="https://user-images.githubusercontent.com/312855/172104574-72d80b9c-7872-461a-b589-8233b377c4a1.png">
```
Twig\Error\RuntimeError:
Key "normalizer" for array with keys "data, dataType, type, format, time, context, normalization, encoding, encoder" does not exist.

  at /redacted/vendor/symfony/web-profiler-bundle/Resources/views/Collector/serializer.html.twig:191
```

**2️⃣ The second thing**:
Create a simple controller and a class:
```php
#[Route('/test2')]
public function test2(SerializerInterface $serializer)
{
    $serializer->deserialize(
        json_encode(['test' => 'string'], JSON_THROW_ON_ERROR),
        Foo::class,
        'json',
    );
}
}

class Foo {
    public array $test = [];
}
```
Open /test2 with a browser and notice that profiler panel is `Loading...` (because profiler data wasn't saved):
<img width="1403" alt="image" src="https://user-images.githubusercontent.com/312855/172104857-019762f1-7d99-42b1-8bb6-81b7f0e1811a.png">

I'm not sure how to get a correct trace for the second case, but there are some problems in the collector:
```
Undefined array key "data" at vendor/symfony/serializer/DataCollector/SerializerDataCollector.php:183 
Undefined array key "format" at vendor/symfony/serializer/DataCollector/SerializerDataCollector.php:186
Undefined array key "time" at vendor/symfony/serializer/DataCollector/SerializerDataCollector.php:188
Undefined array key "context" at vendor/symfony/serializer/DataCollector/SerializerDataCollector.php:189
Undefined array key "method" at vendor/symfony/serializer/DataCollector/SerializerDataCollector.php:224
```

### How to reproduce

Please see the description

**Comment by norkunas** [comment]
It's already fixed, see #46553

**Comment by b1rdex** [comment]
> It's already fixed, see #46553

@norkunas thanks, the first part of the issue is obsolete then.

**Comment by b1rdex** [comment]
The second part is fixed in https://github.com/symfony/symfony/pull/46527
